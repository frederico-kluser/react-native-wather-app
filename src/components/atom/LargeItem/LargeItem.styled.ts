import styled from 'styled-components/native';

const Container = styled.View`
  margin-bottom: 6%;
`;

export const DayOfWeek = styled.Text``;

export const WeatherIcon = styled.Image`
  position: absolute;
  left: 47%;
  height: 32px;
  width: 32px;
`;

export const MaxTemperature = styled.Text`
  position: absolute;
  left: 80%;
  font-size: 17px;
`;

export const MinTemperature = styled.Text`
  position: absolute;
  right: 0%;
  font-size: 17px;
`;

export default Container;
