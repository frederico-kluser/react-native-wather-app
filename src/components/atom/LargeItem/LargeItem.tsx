import React from 'react';
import convertKelvinToCelsius from '../../../helpers/conversor';
import {weatherIcons} from '../../../styles/icons';
import Container, {
  DayOfWeek,
  MaxTemperature,
  MinTemperature,
  WeatherIcon,
} from './LargeItem.styled';

const LargeItem = ({icon, min, max, text}: any) => {
  const date = new Date(parseInt(`${text}000`));
  const daysOfWeek = [
    'segunda',
    'terça',
    'quarta',
    'quinta',
    'sexta',
    'sábado',
    'domingo',
  ];

  return (
    <Container>
      <DayOfWeek>{daysOfWeek[date.getDay()]}</DayOfWeek>
      <WeatherIcon source={weatherIcons[icon]} />
      <MaxTemperature>{convertKelvinToCelsius(min)}°</MaxTemperature>
      <MinTemperature>{convertKelvinToCelsius(max)}°</MinTemperature>
    </Container>
  );
};

export default LargeItem;
