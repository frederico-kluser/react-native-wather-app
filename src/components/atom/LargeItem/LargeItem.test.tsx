import React from 'react';
import renderer from 'react-test-renderer';
import LargeItem from './LargeItem';

test('renders correctly', () => {
  const tree = renderer.create(<LargeItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
