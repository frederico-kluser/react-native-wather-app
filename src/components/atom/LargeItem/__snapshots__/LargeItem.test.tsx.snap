// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": "6%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {},
      ]
    }
  >
    Monday
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/sunStorm.png",
      }
    }
    style={
      Array [
        Object {
          "height": 32,
          "left": "47%",
          "position": "absolute",
          "width": 32,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 17,
          "left": "80%",
          "position": "absolute",
        },
      ]
    }
  >
    32°
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 17,
          "position": "absolute",
          "right": "0%",
        },
      ]
    }
  >
    22°
  </Text>
</View>
`;
